src/tir/transforms/compact_buffer_region.cc

```diff
@@ -481,47 +520,67 @@ class BufferCompactor : public StmtExprMutator {
     return std::move(block);
   }
 
-  Array<Buffer> RewriteAllocBuffer(const Array<Buffer>& buffers) {
-    Array<Buffer> result;
-    result.reserve(buffers.size());
-    for (const Buffer& buffer : buffers) {
-      auto it = buffer_info_.find(buffer);
-      ICHECK(it != buffer_info_.end());
-      BufferAllocInfo& info = it->second;
-      Array<PrimExpr> shape;
-      shape.reserve(info.region.size());
-      for (const Range& range : info.region) {
-        shape.push_back(range->extent);
-      }
-      Array<PrimExpr> strides;
-      if (info.dim_aligns.size()) {
-        ICHECK(info.dim_aligns.size() == shape.size());
-        strides.resize(shape.size());
-        PrimExpr stride = make_const(shape[0].dtype(), 1);
-        for (size_t i = shape.size(); i != 0; --i) {
-          size_t dim = i - 1;
-          if (info.dim_aligns[dim].align_factor != 0) {
-            PrimExpr factor = make_const(stride.dtype(), info.dim_aligns[dim].align_factor);
-            PrimExpr offset = make_const(stride.dtype(), info.dim_aligns[dim].align_offset);
-            stride = stride + indexmod(factor + offset - indexmod(stride, factor), factor);
-          }
-          strides.Set(dim, stride);
-          stride = stride * shape[dim];
+  Stmt VisitStmt_(const DeclBufferNode* op) final {
+    Buffer new_buffer = RewriteAllocBuffer(op->buffer);
+    auto n = CopyOnWrite(op);
+    n->buffer = std::move(new_buffer);
+    n->body = VisitStmt(op->body);
+    return DeclBuffer(n);
+  }
+
+  Stmt VisitStmt_(const AllocateNode* op) final {
```

Comment:
Should we also support `AllocateConstNode`?

Reply:
I am not sure whether we are allowed to trim the const data, currently the const data is left as it is.  I am open to try it if we think it is neccesary :)

Reply:
Good point.  The potential issue I was picturing was an `AllocateConst`/`DeclBuffer` pair, where only the `DeclBuffer` would be rewritten.  Now that the `Allocate` is used to identify locations that should be rewritten, rather than the `DeclBuffer`, this case can no longer occur.